packages/dev/core/src/Lights/rectAreaLight.ts

```diff
@@ -0,0 +1,139 @@
+
+/**
+ * A rectangular area light defined by an unique point in world space, a width and a height.
+ * The light is emitted from the rectangular area in the -Z direction.
+ */
+export class RectAreaLight extends AreaLight {
+    protected _width: Vector3;
+    protected _height: Vector3;
+
+    /**
+     * Rect Area Light width.
+     */
+    @serialize()
+    public get width(): number {
+        return this._width.x;
+    }
+    /**
+     * Rect Area Light width.
+     */
+    public set width(value: number) {
+        this._width.x = value;
+    }
+
+    /**
+     * Rect Area Light height.
+     */
+    @serialize()
+    public get height(): number {
+        return this._height.y;
+    }
+    /**
+     * Rect Area Light height.
+     */
+    public set height(value: number) {
+        this._height.y = value;
+    }
+
+    /**
+     * Creates a rectangular area light object.
+     * Documentation : https://doc.babylonjs.com/features/featuresDeepDive/lights/lights_introduction
+     * @param name The friendly name of the light
+     * @param position The position of the area light.
+     * @param width The width of the area light.
+     * @param height The height of the area light.
+     * @param scene The scene the light belongs to
+     */
+    constructor(name: string, position: Vector3, width: number, height: number, scene?: Scene) {
+        super(name, position, scene);
+        this.position = position;
+        this._width = new Vector3(width, 0, 0);
+        this._height = new Vector3(0, height, 0);
+    }
+
+    /**
+     * Returns the string "RectAreaLight"
+     * @returns the class name
+     */
+    public override getClassName(): string {
+        return "RectAreaLight";
+    }
+
+    /**
+     * Returns the integer 4.
+     * @returns The light Type id as a constant defines in Light.LIGHTTYPEID_x
+     */
+    public override getTypeID(): number {
+        return Light.LIGHTTYPEID_RECT_AREALIGHT;
+    }
+
+    protected _buildUniformLayout(): void {
+        this._uniformBuffer.addUniform("vLightData", 4);
+        this._uniformBuffer.addUniform("vLightDiffuse", 4);
+        this._uniformBuffer.addUniform("vLightSpecular", 4);
+        this._uniformBuffer.addUniform("vLightWidth", 4);
+        this._uniformBuffer.addUniform("vLightHeight", 4);
+        this._uniformBuffer.addUniform("shadowsInfo", 3);
+        this._uniformBuffer.addUniform("depthValues", 2);
+        this._uniformBuffer.create();
+    }
+
+    protected _pointTransformedPosition: Vector3;
+    protected _pointTransformedWidth: Vector3;
+    protected _pointCTransformedHeight: Vector3;
+
+    protected _computeTransformedInformation(): boolean {
+        if (this.parent && this.parent.getWorldMatrix) {
+            if (!this._pointTransformedPosition) {
+                this._pointTransformedPosition = Vector3.Zero();
+                this._pointTransformedWidth = Vector3.Zero();
+                this._pointCTransformedHeight = Vector3.Zero();
+            }
```

Comment:
Why lazy create these? Can they just be created in the constructor to avoid additional `if` check? If so, the properties should be readonly.